### PR TITLE
fix(netlify-cms-core): fields metadata

### PR DESCRIPTION
**Pull request by barthc**
**Summary**
This PR would prepend field's name for field's metadata. Currently, the kitchen sink `relationKitchenSinkPost` widget custom preview only works for the first top-level widget but does not work for the other two nested `relationKitchenSinkPost` widget. This PR fixes the issue. Plus the fix for  #1982 depending on this PR.

<!--
Demonstrate the code is solid.
Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.
--

### CODE REVIEW

File: website/content/docs/customization.md [hunk 1]
```diff
@@ -168,12 +168,12 @@ Registers a template for a folder collection or an individual file in a file col
...
     ### Accessing Metadata
-    Preview Components also receive an additional prop: `fieldsMetaData`. It contains aditional information (besides the plain plain textual value of each field) that can be useful for preview purposes. For example, the Relation widget passes the whole selected relation data in `fieldsMetaData`.
+    Preview Components also receive an additional prop: `metadata`. It contains aditional information (besides the plain plain textual value of each field) that can be useful for preview purposes. For example, the Relation widget passes the whole selected relation data in `metadata`.
     ```js
     export default class ArticlePreview extends React.Component {
       render() {
-        const {entry, fieldsMetaData} = this.props;
-        const author = fieldsMetaData.getIn(['authors', data.author]);
+        const {entry, metadata} = this.props;
+        const author = metadata.getIn(['authors', data.author]);
```
[the remark this answers is not in the document]
Comment (barthc) [reply website/content/docs/customization.md]: @erquhart  reverted back to `fieldsMetaData`.

**Comment by erquhart** [comment]
@barthc I just tried out the relation widgets in the kitchen sink from [this PR preview](https://deploy-preview-2011--cms-demo.netlify.com/#/collections/kitchenSink/entries/a-big-entry-with-all-the-things) and from the [current cms demo](https://cms-demo.netlify.com/#/collections/kitchenSink/entries/a-big-entry-with-all-the-things), and the previews for both behave the same - not present on load, but shows when you edit. Are you seeing anything different?

**Comment by barthc** [comment]
@erquhart  that is a known issue #1541. The simple fix for that is to allow the preview to update on `fieldsMetaData` change.

**Comment by erquhart** [comment]
I wasn't able to reproduce the issue you mentioned in the OP, but I spent a little more time and found it 👍👍

**Pull request merged**